### PR TITLE
track submissions fixes

### PR DESCRIPTION
**Pull request by timabilov**

### CODE REVIEW

File: infinite_games/base/validator.py [hunk 1]
```diff
@@ -496,7 +496,7 @@ def send_event_scores(self, miner_scores=None):
         else:
             bt.logging.debug('*** Grafana logs sent')
 
-    @backoff.on_exception(backoff.expo, Exception, max_tries=6)
+    # @backoff.on_exception(backoff.expo, Exception, max_tries=6)
```
Comment (MadMedian) [inline infinite_games/base/validator.py:499]: was this removed on purpopse? I think exponential backoff could be good here

File: infinite_games/base/validator.py [hunk 2]
```diff
@@ -532,12 +532,13 @@ def send_interval_data(self, miner_data):
                 )
                 if not res.status_code == 200:
                     bt.logging.warning(f'Error processing submission {res.content}')
+                    return False
                 else:
                     return True
-                time.sleep(1)
```
Comment (MadMedian) [inline infinite_games/base/validator.py:537]: not sure about the intention here. If the request was not successful, why do we return False instead of raising an error?
we could simply check `res.raise_for_status()` and then log the error in except

File: infinite_games/events/base.py [hunk 3]
```diff
@@ -756,7 +756,7 @@ def mark_event_as_exported(self, pe: ProviderEvent) -> bool:
         conn.close()
         return False
 
-    def mark_submissions_as_exported(self) -> bool:
+    def mark_submissions_as_exported(self, interval_start_minutes: int, previous_interval_start_minutes) -> bool:
```
Comment (MadMedian) [inline infinite_games/events/base.py:759]: I would rather use backoff than custom code for retries

File: infinite_games/base/validator.py (hunk 2 above)
Comment (timabilov) [reply infinite_games/base/validator.py:537]: @MadMedian to differentiate the other exceptions basically, although we can create exception, i think it is enough for now

File: neurons/validator.py [hunk 4]
```diff
@@ -120,11 +121,17 @@ async def send_interval_stats(self):
...
-                await asyncio.sleep(4)
-            self.event_provider.mark_submissions_as_exported()
+            try:
+                async for metrics in chunk_metrics:
+                    intervals = [metric[3] for metric in metrics]
+                    if not self.send_interval_data(miner_data=metrics):
+                        bt.logging.error('Interrupting the stream, due to backend error..')
+                        return
+                    bt.logging.info(f'chunk submissions exported {len(metrics)} intervals: {set(intervals)}')
+                    await asyncio.sleep(4)
```
Comment (MadMedian) [inline neurons/validator.py:131]: this looks a bit weird, why do we have sleep in asyncio?

File: infinite_games/events/base.py (hunk 3 above)
Comment (timabilov) [reply infinite_games/events/base.py:759]: with backoff we had an hidden logs issue, was hard to debug as mentioned, so here i prefer for explicit logic. and this one is not only for retries. in case if validator is not active or something we have to cover previous intervals

File: neurons/validator.py (hunk 4 above)
Comment (timabilov) [reply neurons/validator.py:131]: thats the purpose of chunks, to not spam backend  with 100k records at once, like a cooldown.

File: infinite_games/base/validator.py (hunk 1 above)
Comment (MadMedian) [reply infinite_games/base/validator.py:499]: > it was creating a lot of problems by hiding logs recently

That sounds strange, do you have some example? If you log in the except branch before reraising the error, do you miss the logs? I see in the documentation some instructions for logging
https://pypi.org/project/backoff/
Also has some arg for additional logic on backoff `on_backoff=backoff_hdlr`

> because we will retry same intervals from logic anyway
if that retry logic was not unit tested, I would very much prefer using stuff from the standard or popular libraries

File: infinite_games/base/validator.py (hunk 2 above)
Comment (MadMedian) [reply infinite_games/base/validator.py:537]: ok, let's log the errors where they happen, because it will be hard to follow.

File: infinite_games/events/base.py (hunk 3 above)
Comment (MadMedian) [reply infinite_games/events/base.py:759]: is this explicit logic unit tested? because otherwise I think it is preferably to use code from standard libraries

File: neurons/validator.py (hunk 4 above)
Comment (MadMedian) [reply neurons/validator.py:131]: ok, is 100k a lot or very little? and why 4 seconds and not 1 or 10...or simply check if the backend is available?

**Review by MadMedian — APPROVED** [review]
we can fix the comments in the following PRs
